File: aizu/Introduction-to-Algorithms-and-Data-Structures/ALDS1_1_C_PrimeNumbers/ALDS1_1_C.c
#include <stdio.h>
#include <math.h>
#define N_MAX 10000

int isPrime(int x)
{
  int i;
  if (x <= 3) {
    return 1;
  } else if (x % 2 == 0) {
    return 0;
  } else {
    for (i = 3; i <= sqrt(x); i = i + 2) {
      if (x % i == 0) {
        return 0;
      }
    }
    return 1;
  }
}

int main(int argc, char *argv[])
{
  int i, j, n, x;
  int answer = 0;
  scanf("%d", &n);
  for (i = 0; i < n; i++) {
    scanf("%d", &x);
    answer = answer + isPrime(x);
  }
  printf("%d\n", answer);
  return 0;
}

File: aizu/Introduction-to-Algorithms-and-Data-Structures/ALDS1_1_C_PrimeNumbers/ALDS1_1_C.go
package main

import (
	"bufio"
	"fmt"
	"math"
	"os"
	"strconv"
)

func isPrime(x int) bool {
	if x == 0 {
		return true
	} else if x == 1 {
		return true
	} else if x == 2 {
		return true
	} else if x%2 == 0 {
		return false
	} else {
		y := math.Sqrt(float64(x))
		for i := 3; float64(i) <= y; i = i + 2 {
			if x%i == 0 {
				return false
			}
		}
		return true
	}
}

func prime(xs []int) {
	count := 0
	for _, x := range xs {
		if isPrime(x) {
			count = count + 1
		}
	}
	fmt.Println(count)
}

func nextInt(sc *bufio.Scanner) int {
	sc.Scan()
	s := sc.Text()
	n, err := strconv.Atoi(s)
	if err != nil {
		panic(err)
	}
	return n
}

func main() {
	sc := bufio.NewScanner(os.Stdin)
	sc.Split(bufio.ScanWords)

	n := nextInt(sc)

	xs := make([]int, n)
	for i := 0; i < n; i++ {
		xs[i] = nextInt(sc)
	}
	prime(xs)
}
